### PR TITLE
DDF-4091 mapped documentation module directories to docs outline

### PR DESCRIPTION
**Pull request by ricklarsen**
#### What does this PR do?

reorders the directory structure of the docs content to map to the primary documentation outline for ease of maintainability.

#### Who is reviewing it? 
@mcalcote @ahoffer @austinsteffes @garrettfreibott @emmberk 

#### Ask 2 committers to review/merge the PR and tag them here.

@clockard
@lessarderic
@shaundmorris

#### How should this be tested?

builds successfully; no changes made to content
(will need manual confirmation of build on Windows)

#### What are the relevant tickets?
[DDF-4091](https://codice.atlassian.net/browse/DDF-4091)

#### Checklist:
- [x] Documentation Updated
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

**Comment by ricklarsen** [comment]
build now

File: .gitignore [hunk 1]
```diff
@@ -19,41 +19,10 @@ atlassian-ide-plugin.xml
...
-distribution/docs/src/main/resources/content/_sources/.asciidoctor/
-distribution/docs/src/main/resources/content/_sources/embedded-solr-diagram.png
-distribution/docs/src/main/resources/content/_sources/http-solr-external-diagram.png
-distribution/docs/src/main/resources/content/_sources/http-solr-internal-diagram.png
-distribution/docs/src/main/resources/content/_sources/solrcloud-diagram.png
-distribution/docs/src/main/resources/content/_transformers/.asciidoctor
-distribution/docs/src/main/resources/content/_transformers/catalog_architecture_transformers.png
-distribution/docs/src/main/resources/content/_transformers/transformer_communication_diagram.png
+distribution/docs/src/main/resources/content/*/*/.asciidoctor
+distribution/docs/src/main/resources/content/*/*/*.png
```
Comment (emmberk) [inline .gitignore]: Commenting on this file so I can start a thread.

✏️ I see 2 `managing-intro.adoc` files. One under `_introduction` and one under `_managing`.
✏️ Why is _developing split into _devComponents and _devGuidelines when the outline structure is 
```
Developing
+ 18. Catalog Framework API
+ 19. Transformers
+ 20. Catalog Plugins
21. Operations
+ 22. Resources
+ 23. Queries
+ 24. Metrics
+ 25. Action Framework
26. Asynchronous Processing Framework
+ 27. Migration API
+ 28. Security Framework
+ 29. Developing DDF Components
+ 30. Development Guidelines
```
✏️ `_reference` appears to contain all of the content for the appendices. Should `_metadataReference` be under `_reference`? Should `_tables` be moved from `_reference`?

Comment (ricklarsen) [reply .gitignore]: I guess I jumped the gun on following the new outline. I'll have PRs up soon to break up the developing sections into an `architecture` reference and a more task-oriented `developing section. The appendices are also getting separated out into a standalone `reference` guide and a metadata guide.

**Comment by ricklarsen** [comment]
rebased to clear conflicts, rerunning build now

**Comment by ricklarsen** [comment]
previous build: yarn errors, rerunning build now

**Pull request merged**